v2/inbound_transport_middleware.go

```diff
@@ -35,15 +37,25 @@ import (
 // UnaryInboundTransportMiddleware is re-used across requests and MAY be called multiple
 // times for the same request.
 type UnaryInboundTransportMiddleware interface {
+	Name() string
 	Handle(ctx context.Context, req *Request, reqBuf *Buffer, handler UnaryTransportHandler) (*Response, *Buffer, error)
 }
 
+// UnaryInboundTransportMiddlewareProvider is a registry of pre-configured UnaryInboundTransportMiddleware.
+type UnaryInboundTransportMiddlewareProvider interface {
+	Middleware(name string) (UnaryInboundTransportMiddleware, bool)
```

Comment:
It's not immediately clear why the signature includes a bool - a comment would be useful here